File: android/src/main/kotlin/io/sariska/sariska_media_transport/ConnectionManager.kt
package io.sariska.sariska_media_transport
import io.sariska.sdk.Connection

class ConnectionManager(
        private val emit: (action: String) -> Unit
) : Connection("") {
    lateinit var connection: Connection

    fun createConnection(params: Map<String, *>) {
        connection = Connection(params["token"] as? String)
    }

    override fun newConnectionMessage(action: String?) {
        if (action != null) {
            emit(action)
        }
    }

    fun destroy() {
        connection.disconnect()
    }

    override fun connect() {
        connection.connect()
    }

    override fun disconnect() {
        connection.disconnect()
    }

    override fun addFeature() {
        connection.addFeature()
    }

    override fun removeFeature() {
        connection.removeFeature()
    }

    fun release(){
        connection.disconnect();
    }
}

File: ios/Classes/ConnectionManager.swift
import Foundation
import sariska

class ConnectionManager: NSObject, Connection {
    private var emitter: (_ action: String, _ m: [String: Any?]?) -> Void
	  let connection: Connection?

    init(_ emitter: @escaping (_ action: String, _ m: [String: Any?]?) -> Void) {
        self.emitter = emitter
    }

    func Release() {
        connection = nil
    }

    @objc func createConnection(_ params: NSDictionary) {
        connection = Connection(params["token"] as! NSString)
    }

    @objc func connect() {
        connection.connect()
    }

    @objc func disconnect() {
        connection.disconnect()
    }

    @objc func addFeature() {
        connection.addFeature()
    }

    @objc func removeFeature() {
        connection.removeFeature()
    }

}